policies/redis/producer/streams.go

```diff
@@ -28,6 +36,94 @@ type eventStore struct {
 	logger *zap.Logger
 }
 
+func (e eventStore) RemovePolicy(ctx context.Context, token string, policyID string) error {
+	if err := e.svc.RemovePolicy(ctx, token, policyID); err != nil {
+		return err
+	}
+
+	datasets, err := e.svc.ListDatasetsByPolicyIDInternal(ctx, policyID, token)
+	if err != nil {
+		return err
+	}
+
+	var groupsIDs []string
+	var ownerID string
+	for _, ds := range datasets {
+		ownerID = ds.MFOwnerID
+		groupsIDs = append(groupsIDs, ds.AgentGroupID)
+	}
+
+	event := removePolicyEvent{
+		id:       policyID,
+		ownerID:  ownerID,
+		groupIDs: strings.Join(groupsIDs, ","),
+	}
+	record := &redis.XAddArgs{
+		Stream:       streamID,
+		MaxLenApprox: streamLen,
+		Values:       event.Encode(),
+	}
+	err = e.client.XAdd(ctx, record).Err()
+	if err != nil {
+		e.logger.Error("error sending event to event store", zap.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+func (e eventStore) ListDatasetsByPolicyIDInternal(ctx context.Context, policyID string, token string) ([]policies.Dataset, error) {
+	return e.svc.ListDatasetsByPolicyIDInternal(ctx, policyID, token)
+}
+
+func (e eventStore) EditPolicy(ctx context.Context, token string, pol policies.Policy, format string, policyData string) (policies.Policy, error) {
+	res, err := e.svc.EditPolicy(ctx, token, pol, format, policyData)
+	if err != nil {
+		return policies.Policy{}, err
+	}
+
+	datasets, err := e.svc.ListDatasetsByPolicyIDInternal(ctx, res.ID, token)
+	if err != nil {
+		return policies.Policy{}, err
+	}
+
+	var groupsIDs []string
+	for _, ds := range datasets {
+		groupsIDs = append(groupsIDs, ds.AgentGroupID)
+	}
+
+	p, err := e.svc.ViewPolicyByID(ctx, token, pol.ID)
+	if err != nil {
+		return policies.Policy{}, err
+	}
+	pol.Backend = p.Backend
+	pol.MFOwnerID = p.MFOwnerID
+
+	err = validatePolicyBackend(&pol, format, policyData)
+	if err != nil {
+		return policies.Policy{}, err
+	}
+
+	event := updatePolicyEvent{
+		id:       pol.ID,
+		ownerID:  pol.MFOwnerID,
+		groupIDs: strings.Join(groupsIDs, ","),
+		policy:   pol.Policy,
```

Comment:
It seems like we aren't using this right now because the fleet service is making an RPC call back to policies to retrieve it during an edit. If we are going to do that for now, then let's remove this policy from redis because they could be large and waste CPU/memory.

Reply:
Ok, done.